File: MicroM/micromlib/src/Entity/DefaultColumns.ts
import { EntityColumn } from "./EntityColumn";
import { CommonFlags as c, EntityColumnFlags as f } from "./EntityColumn.types";
import { ColumnsObject } from "./EntityColumnCollection.types";

export const SYSTEM_COLUMNS_NAMES = ['dt_inserttime', 'dt_lu', 'vc_webinsuser', 'vc_webluuser', 'vc_insuser', 'vc_luuser'];

export interface IDefaultColumns extends ColumnsObject {
    dt_inserttime: EntityColumn<Date>,
    dt_lu: EntityColumn<Date>,
    vc_webinsuser: EntityColumn<string>,
    vc_webluuser: EntityColumn<string>,
    vc_insuser: EntityColumn<string>,
    vc_luuser: EntityColumn<string>,
}

export const DefaultColumnsNames = ['dt_inserttime', 'dt_lu', 'vc_webinsuser', 'vc_webluuser', 'vc_insuser', 'vc_luuser'];

export const DefaultColumns = (): IDefaultColumns =>
(
    {
        dt_inserttime: new EntityColumn<Date>({ name: 'dt_inserttime', type: 'datetime', flags: f.None }),
        dt_lu: new EntityColumn<Date>({ name: 'dt_lu', type: 'datetime', flags: c.Edit }),
        vc_webinsuser: new EntityColumn<string>({ name: 'vc_webinsuser', type: 'varchar', length: 255, flags: c.Edit }),
        vc_webluuser: new EntityColumn<string>({ name: 'vc_webluuser', type: 'varchar', length: 255, flags: f.None }),
        vc_insuser: new EntityColumn<string>({ name: 'vc_insuser', type: 'varchar', length: 255, flags: f.None }),
        vc_luuser: new EntityColumn<string>({ name: 'vc_luuser', type: 'varchar', length: 255, flags: f.None }),
    }
)

File: MicroM/core/Data/DefaultColumns.cs
namespace MicroM.Data;

internal class DefaultColumns
{
    public readonly Column<DateTime> dt_inserttime = DefaultColumnsFactory.dt_inserttime();
    public readonly Column<DateTime> dt_lu = DefaultColumnsFactory.dt_lu();
    public readonly Column<string> vc_webinsuser = DefaultColumnsFactory.vc_webinsuser();
    public readonly Column<string> vc_webluuser = DefaultColumnsFactory.vc_webluuser();
    public readonly Column<string> vc_insuser = DefaultColumnsFactory.vc_insuser();
    public readonly Column<string> vc_luuser = DefaultColumnsFactory.vc_luuser();
    public readonly Column<string> webusr = DefaultColumnsFactory.webusr();

    public readonly static string[] SystemNames = {
            nameof(dt_inserttime),
            nameof(dt_lu),
            nameof(vc_webinsuser),
            nameof(vc_webluuser),
            nameof(vc_insuser),
            nameof(vc_luuser),
            nameof(webusr)
        };


}
